gravitee-apim-console-webui/src/management/api/documentation-v4/documentation-metadata/api-documentation-v4-metadata.component.ts

```diff
@@ -28,12 +31,23 @@ export class ApiDocumentationV4MetadataComponent implements OnInit {
   metadataSaveServices: MetadataSaveServices;
   description: string;
 
-  constructor(private readonly apiService: ApiService, private readonly activatedRoute: ActivatedRoute) {}
+  constructor(
+    private readonly apiService: ApiService,
+    private readonly apiMetadataV2Service: ApiMetadataV2Service,
+    private readonly activatedRoute: ActivatedRoute,
+  ) {}
 
   ngOnInit() {
     this.metadataSaveServices = {
       type: 'API',
-      list: () => this.apiService.listMetadata(this.activatedRoute.snapshot.params.apiId),
+      paginate: true,
+      list: (searchMetadata) =>
+        this.apiMetadataV2Service.searchApiMetadata(this.activatedRoute.snapshot.params.apiId, searchMetadata).pipe(
+          map((resp) => ({
+            data: resp.data?.map((metadata) => ({ ...metadata } as Metadata)),
+            totalResults: resp.pagination?.totalCount ?? resp.data?.length ?? 0,
```

Comment:
🤔 totalCount are normally allays set no ? You keep resp.data?.length as safe guard ?

Reply:
True, it can only be null if the list is empty.